Actions/RunPipeline/RunPipeline.ps1

```diff
@@ -119,6 +124,111 @@ try {
         exit
     }
 
+    $buildArtifactFolder = Join-Path $projectPath ".buildartifacts"
+    New-Item $buildArtifactFolder -ItemType Directory | Out-Null
+
+    if ($baselineWorkflowSHA -and $baselineWorkflowRunId -ne '0' -and $settings.incrementalBuilds.mode -eq 'modifiedApps') {
+        # Incremental builds are enabled and we are only building modified apps
+        $headSHA = git rev-parse HEAD
+        Write-Host "Current HEAD is $headSHA"
+        git fetch origin $baselineWorkflowSHA | Out-Host
+        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to fetch baseline SHA $baselineSHA" }
+        Push-Location $ENV:GITHUB_WORKSPACE
+        Write-Host "git diff --name-only $baselineWorkflowSHA $headSHA"
+        $modifiedFiles = @(git diff --name-only $baselineWorkflowSHA $headSHA | ForEach-Object { "$_".Replace('/', [System.IO.Path]::DirectorySeparatorChar) })
+        if ($LASTEXITCODE -ne 0) { $host.SetShouldExit(0); throw "Failed to diff baseline SHA $baselineSHA with current HEAD $headSHA" }
+        Pop-Location
+        Write-Host "$($modifiedFiles.Count) modified file(s)"
+        if ($modifiedFiles.Count -gt 0) {
+            foreach($modifiedFile in $modifiedFiles) {
+                Write-Host "- $modifiedFile"
+            }
+        }
+        $buildAll = Get-BuildAllApps -baseFolder $baseFolder -project $project -modifiedFiles $modifiedFiles
+        if (!$buildAll) {
+            $skipFolders = @()
```

Comment:
Consider extracting a separate `Download-AppArtifacts` function to run the logic here.

Reply:
Considering that this probably is the last large feature implementation in RunPipeline and that the new build mechanism, which doesn't use BcContainerHelper will be re-written, I would rather keep this as is.

Reply:
Could be a while until we get to that point. It's also easier to refactor/rewrite code when it's contained within functions, modules etc.

Reply:
This code is not going to be refactored. The functionality needs to be rewritten

Reply:
No, RunPipeline is going to be written from scratch (I think).
But... - I have moved the function out.